src/eureka/lib/apphot.py

```diff
@@ -1,467 +1,74 @@
-    ret[:] = np.nan
-
-    # set error status to 'good' = 0
-    ret[stat] = 0
+    # set initial error status to 'good' = 0
     status = 0
-
     # bit flag definitions
     statnan = 2 ** 0
     statbad = 2 ** 1
     statap = 2 ** 2
     statsky = 2 ** 3
 
-    # internal skyfrac, so we don't set it in caller
-    iskyfrac = skyfrac
-
-    # Check inputs
-    sz = np.shape(image)
-
-    if not nochecks:
-        if np.ndim(image) != 2:
-            # FINDME: raise exceptions instead
-            print('image must be a 2D array')
-            # break
-            return ret[np.where(retidx)]
-
-        # Default mask: only non-finite values are bad
-        if mask is None:
-            mask = ~np.isfinite(image)
-
-        if np.ndim(mask) != 2:
-            print('mask must be a 2D array')
-            return ret[np.where(retidx)]
-
-        if (np.shape(image) != np.shape(mask)):
-            print('image and mask sizes differ')
-            return ret[np.where(retidx)]
-
-        if imerr is not None:
-            if (np.shape(image) != np.shape(imerr)):
-                print('image and imerr sizes differ')
-                return ret[np.where(retidx)]
-
-        if (iskyfrac < 0.0) or (iskyfrac > 1.0):
-            print('skyfrac must be in range [0,1]')
-            return ret[np.where(retidx)]
-
-        if expand != np.compat.long(expand) or expand < 1:
-            print('invalid expand')
-            return ret[np.where(retidx)]
-
     # Expand
-    iexpand = int(expand)
-    isz = np.array(sz, dtype=int)+(np.array(sz, dtype=int)-1)*(iexpand-1)
-    ictr = iexpand * np.array(ctr)
-    iphotap = iexpand * photap
-    iskyin = iexpand * skyin
-    iskyout = iexpand * skyout
+    sz = np.shape(image)
+    isz = np.array(sz, dtype=int)+(np.array(sz, dtype=int)-1)*(meta.expand-1)
+    ictr = meta.expand*np.array(ctr)
+    iphotap = meta.expand*meta.photap
+    iskyin = meta.expand*meta.skyin
+    iskyout = meta.expand*meta.skyout
 
     y, x = np.arange(sz[0]), np.arange(sz[1])
     yi, xi = np.linspace(0, sz[0]-1, isz[0]), np.linspace(0, sz[1]-1, isz[1])
-    iimage = i2d.interp2d(image, expand=iexpand, y=y, x=x, yi=yi, xi=xi)
-    imask = i2d.interp2d(mask.astype(float), expand=iexpand, y=y, x=x,
+    iimage = i2d.interp2d(image, expand=meta.expand, y=y, x=x, yi=yi, xi=xi)
+    iimerr = i2d.interp2d(imerr, expand=meta.expand, y=y, x=x, yi=yi, xi=xi)
+    imask = i2d.interp2d(mask.astype(float), expand=meta.expand, y=y, x=x,
                          yi=yi, xi=xi)
-    # Need to convert fractions to booleans
-    imask = imask > 0.5
-    if imerr is not None:
-        iimerr = i2d.interp2d(imerr, expand=iexpand, y=y, x=x, yi=yi, xi=xi)
+    imask = imask > 0.5  # Need to convert fractions back to booleans
 
     # Specify aperture shape function
-    if aperture_shape == "hexagon":
+    if meta.aperture_shape == "hexagon":
         apFunc = di.hex
-    else:
+    elif meta.aperture_shape == "circle":
         apFunc = di.disk
```

Comment:
If we haven't done so already, we should catch the case where the aperture shape isn't hexagon or circle.

Reply:
This has already been caught in s3_meta.py